transports/janus_rabbitmq.c

```diff
@@ -900,8 +936,34 @@ void *janus_rmq_in_thread(void *data) {
 			if(res == AMQP_STATUS_TIMEOUT || res == AMQP_STATUS_SSL_ERROR)
 				continue;
 			JANUS_LOG(LOG_VERB, "Error on amqp_simple_wait_frame_noblock: %d (%s)\n", res, amqp_error_string2(res));
-			break;
+
+			rmq_client->connected = 0;
+
+			/* Try and reconnect */
+			if(rmq_client->rmq_conn && rmq_client->rmq_channel) {
+				amqp_channel_close(rmq_client->rmq_conn, rmq_client->rmq_channel, AMQP_REPLY_SUCCESS);
+				amqp_connection_close(rmq_client->rmq_conn, AMQP_REPLY_SUCCESS);
+				amqp_destroy_connection(rmq_client->rmq_conn);
```

Comment:
according to the docs, destroying the connection will also close the same connection and every channel

Reply:
Fixed - I copied the logic from the event handler so I've updated the logic there too